tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -403,6 +405,53 @@ class GraphString {
   std::vector<OpDescriptor> graph_;
 };
 
+bool IsF8Type(const HloInstruction* instr) {
+  return primitive_util::IsF8Type(instr->shape().element_type());
+}
+
+bool IsScalar(const HloInstruction* instr) {
+  return ShapeUtil::IsScalar(instr->shape());
+}
+
+std::optional<PrimitiveType> IsSaturatingCastToF8(HloInstruction* instr) {
+  HloInstruction *op, *clamp_lower, *clamp_upper;
+  if (Match(instr,
+            m::Convert(
+                &op,
+                m::Clamp(m::Broadcast(m::ConstantScalar(&clamp_lower)), m::Op(),
+                         m::Broadcast(m::ConstantScalar(&clamp_upper))))) &&
+      (op->shape().element_type() == F8E4M3FN &&
+           clamp_lower->literal().IsAllFloat(static_cast<float>(
+               std::numeric_limits<tsl::float8_e4m3fn>::lowest())) &&
+           clamp_upper->literal().IsAllFloat(static_cast<float>(
+               std::numeric_limits<tsl::float8_e4m3fn>::max())) ||
+       op->shape().element_type() == F8E5M2 &&
+           clamp_lower->literal().IsAllFloat(static_cast<float>(
+               std::numeric_limits<tsl::float8_e5m2>::lowest())) &&
+           clamp_upper->literal().IsAllFloat(static_cast<float>(
+               std::numeric_limits<tsl::float8_e5m2>::max())))) {
+    return op->shape().element_type();
+  }
+  return std::nullopt;
+}
+
+// Returns whether the HLO Computation applied by `op` calculates the largest
+// element.
+bool AppliesMaxReduce(HloInstruction* op) {
+  HloComputation* reduce_comp = op->to_apply();
+  HloInstruction* reduce_comp_root = reduce_comp->root_instruction();
+  if (ShapeUtil::IsScalar(op->shape()) &&
+      ShapeUtil::IsScalar(op->operand(1)->shape()) &&
+      op->operand(1)->IsConstant() &&
+      op->operand(1)->literal().GetAsDouble({}) <= 0. &&
+      reduce_comp_root->opcode() == HloOpcode::kMaximum &&
+      reduce_comp_root->operand(0)->opcode() == HloOpcode::kParameter &&
+      reduce_comp_root->operand(1)->opcode() == HloOpcode::kParameter) {
+    return true;
+  }
+  return false;
```

Comment:
No need for the if-statement, you can directly return the expression: `return ShapeUtil::IsScalar(op->shape()) && ...;`